Sim/SimDelphesInterface/src/DelphesSimulation.cpp

```diff
@@ -343,78 +369,452 @@ StatusCode DelphesSimulation::finalize() {
+    double recE   = sqrt(relation.Rec().Core().P4.Px*relation.Rec().Core().P4.Px +
+                         relation.Rec().Core().P4.Py*relation.Rec().Core().P4.Py +
+                         relation.Rec().Core().P4.Pz*relation.Rec().Core().P4.Pz +
+                         relation.Rec().Core().P4.Mass*relation.Rec().Core().P4.Mass);
+    double simE   = sqrt(relation.Sim().Core().P4.Px*relation.Sim().Core().P4.Px +
+                         relation.Sim().Core().P4.Py*relation.Sim().Core().P4.Py +
+                         relation.Sim().Core().P4.Pz*relation.Sim().Core().P4.Pz +
+                         relation.Sim().Core().P4.Mass*relation.Sim().Core().P4.Mass);
+
+    std::cout << "Delphes Track: "
+              << " Id: "       << setw(3)  << j+1
+              << " Pdg: "      << setw(5)  << particle.Core().Type
+              << " Stat: "     << setw(2)  << particle.Core().Status
+              << " Bits: "     << setw(2)  << particle.Core().Bits
+              << " Px: "       << setw(10) << particle.Core().P4.Px
+              << " Py: "       << setw(10) << particle.Core().P4.Py
+              << " Pz: "       << setw(10) << particle.Core().P4.Pz
+              << " E: "        << setw(10) << energy
+              << " M: "        << setw(10) << particle.Core().P4.Mass
+              << " Vx: "       << setw(10) << particle.Core().Vertex.X
+              << " Vy: "       << setw(10) << particle.Core().Vertex.Y
+              << " Vz: "       << setw(10) << particle.Core().Vertex.Z
+              << " RefId: "    << setw(3)  << idRefMCPart+1
+              << " Rel E: "    << setw(10) << simE << " <-> " << recE;
+    std::cout << std::endl;*/
+  } // For - tracks
 }
 
-void DelphesSimulation::ConvertMET(   TObjArray *  Input , fcc::ParticleCollection *  coll  ){
+void DelphesSimulation::ConvertTowers(const TObjArray*  Input,
+                                      fcc::ParticleCollection*   colParticles,
+                                      fcc::MCParticleCollection* colMCParticles,
+                                      fcc::ParticleMCParticleAssociationCollection* ascColParticlesToMC)
+{
+  Candidate* cand = nullptr;
+  for(int j=0; j<Input->GetEntries(); j++) {
 
-  Candidate * cand;
-  for(int j = 0; j < Input->GetEntries(); ++j) {
+    cand = static_cast<Candidate *>(Input->At(j));
+
+    auto particle = colParticles->create();
+
+    auto barePart      = new fcc::BareParticle();
+    barePart->Type     = cand->PID;
+    barePart->Status   = cand->Status;
+    barePart->P4.Px    = cand->Momentum.Px();
+    barePart->P4.Py    = cand->Momentum.Py();
+    barePart->P4.Pz    = cand->Momentum.Pz();
+    barePart->P4.Mass  = cand->Momentum.M();
+    barePart->Charge   = cand->Charge;
+    barePart->Vertex.X = cand->Position.X();
+    barePart->Vertex.Y = cand->Position.Y();
+    barePart->Vertex.Z = cand->Position.Z();
+    particle.Core(*barePart);
+
+    // Test print
```

Comment:
I will put debug() in there. Is there several debug levels defined or just one debug level?